src/main/webapp/app/shared/auth/RecaptchaBoundaryRoute.tsx

```diff
@@ -1,31 +1,51 @@
-import { RECAPTCHA_KEY } from 'app/indexUtils';
 import ErrorBoundaryRoute from 'app/shared/error/error-boundary-route';
+import WindowStore from 'app/store/WindowStore';
+import { computed } from 'mobx';
+import { observer } from 'mobx-react';
 import React from 'react';
 import { RouteProps } from 'react-router-dom';
 import { ContactLink } from '../links/ContactLink';
 
 export interface IRecaptchaBoundaryRoute extends RouteProps {
   isUserAuthenticated: boolean;
+  windowStore: WindowStore;
 }
 
-export const RecaptchaBoundaryRoute = ({
-  isUserAuthenticated,
-  ...props
-}: IRecaptchaBoundaryRoute) => {
-  if (isUserAuthenticated || localStorage.getItem(RECAPTCHA_KEY)) {
-    return <ErrorBoundaryRoute {...props} />;
-  } else {
-    return (
-      <>
-        <div>
-          Please try to refresh your page, if you continue seeing this language
-          please{' '}
-          <ContactLink emailSubject={'Unable to Access the Pages'}>
-            contact us
-          </ContactLink>
-          .
-        </div>
-      </>
-    );
+@observer
+export class RecaptchaBoundaryRoute extends React.Component<
+  IRecaptchaBoundaryRoute
+> {
+  @computed
+  get recaptchaVerifyed() {
```

Comment:
Why are we computing on a variable?

Reply:
This route component should be re-render to show real page component when the first time recaptcha got verified, or the page will only show warning message.